packages/toolpad-app/src/toolpadDataSources/rest/client.tsx

```diff
@@ -271,11 +271,12 @@ const EMPTY_PARAMS: BindableAttrEntries = [];
 
 function QueryEditor({
   globalScope,
-  connectionParams,
+  connectionParams: rawConnectionParams,
   value: input,
   onChange: setInput,
 }: QueryEditorProps<RestConnectionParams, FetchQuery>) {
-  const baseUrl = connectionParams?.baseUrl;
+  const connectionParams = input.attributes.query.value.browser ? null : rawConnectionParams;
+  const baseUrl = input.attributes.query.value.browser ? null : connectionParams?.baseUrl;
```

Comment:
or actually here we can just have `const baseUrl = connectionParams?.baseUrl; ?? null`